### PR TITLE
Add pseudo_dojo and dependencies

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

File: recipes/pseudo_dojo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    - pymatgen
+    - abipy >=0.2.0
+
+test:
+  imports:
+    - pseudo_dojo
+
+about:
+  home: http://www.pseudo-dojo.org/
+  license: GPL-3.0-or-later
```
Comment (saraedum) [inline recipes/pseudo_dojo/meta.yaml:46]: The license file on github says GPL 2.1 or later. Are you sure about this one?

Comment (saraedum) [reply recipes/pseudo_dojo/meta.yaml:46]: (either case that's a bit odd since it's using abipy which is 2.0 only.)

Comment (jan-janssen) [reply recipes/pseudo_dojo/meta.yaml:46]: To my knowledge GPL 2.1 is not an official SPDX license, so I chose the next higher one. If you recommend to use `GPL-2.0-or-later` I am fine with that as well.

Comment (saraedum) [reply recipes/pseudo_dojo/meta.yaml:46]: What's GPL 2.1 actually? I only know of LGPL 2.1.

File: recipes/atomicfile/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+test:
+  imports:
+    - atomicfile
+
+about:
+  home: https://github.com/sashka/atomicfile-py
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE.rst
```
Comment (saraedum) [inline recipes/atomicfile/meta.yaml:32]: If you don't mind you could ask upstream whether they could include the license in the future. Not terribly important but they might not be aware of it.

Comment (jan-janssen) [reply recipes/atomicfile/meta.yaml:32]: Sure, I can open a pull request.

File: recipes/pseudo_dojo/pymatgen.patch [hunk 3]
```diff
@@ -0,0 +1,21 @@
+From d35a258f077621d2ff55576476e9d6a8a676d5b8 Mon Sep 17 00:00:00 2001
+From: gmatteo <gmatteo@gmail.com>
+Date: Thu, 17 Oct 2019 19:58:40 +0200
+Subject: [PATCH] drop support for py2, remove six, migrate to flowtk API
```
Comment (saraedum) [inline recipes/pseudo_dojo/pymatgen.patch:4]: Is this a patch from upstream? This subject line does not seem like what's happening in that patch.

File: recipes/pseudo_dojo/meta.yaml (hunk 1 above)
Comment (jan-janssen) [reply recipes/pseudo_dojo/meta.yaml]: Sorry, you are right. It is LGPL-2.1-or-later

File: recipes/atomicfile/meta.yaml (hunk 2 above)
Comment (jan-janssen) [reply recipes/atomicfile/meta.yaml:32]: I created the patch 

```
From 7c1f3b35fe0e09835ce86508f7fcdfb18a252a05 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sat, 16 May 2020 17:55:34 +0200
Subject: [PATCH] include license in pip Package

---
 MANIFEST.in | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 MANIFEST.in

diff --git a/MANIFEST.in b/MANIFEST.in
new file mode 100644
index 0000000..43a950e
--- /dev/null
+++ b/MANIFEST.in
@@ -0,0 +1 @@
+include LICENSE.rst
```

Unfortunately the repository is archived/ read-only so I can not push it.

Comment (saraedum) [reply recipes/atomicfile/meta.yaml:32]: I see. Nevermind.

File: recipes/pseudo_dojo/pymatgen.patch (hunk 3 above)
Comment (jan-janssen) [reply recipes/pseudo_dojo/pymatgen.patch:4]: The patch included more changes, I focused on the parts required for pymatgen compatibility. I hope they soon release a new version.

**Pull request merged**